plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/configuration/DatabaseMetadata.java

```diff
@@ -33,15 +37,42 @@
   private static ObjectMapper MAPPER = new ObjectMapper();
   private final int version;
 
+  private Optional<Integer> privacyVersion;
```

Comment:
What is the motivation for adding the privacy version to the existing metadata rather than creating its own (as privacy does a separate RocksDB file)?

Reply:
Yes, that is an option. I felt the current approach does not pollute the folder with unnecessary files.

Reply:
>  current approach does not pollute the folder with unnecessary files.
That is true ...but:
1. Data of public and private state are separated (stored in different files).
2. In this PR, the code has a much greater delineation between the handling of public and private (interfaces and dedicated privacy classes).
3. An extra file is not the end of the world (as placing files in sub-folders can always be leveraged for organisation).
4. Precedence; today there are only two buckets for storing data, if/when more are needed this will be the approach.

What are you thoughts @iikirilov?

Reply:
right now this meets my needs for #124 and it is well tested - I will likely need to revisit it because it requires major refactoring. There is one class that handles creation of db, reading version and writing the version.